### PR TITLE
Validate that a metric only selects from a single source/transform

### PR DESCRIPTION
**Pull request by samredai**
### Summary

Although we validate that a metric has a single expression that includes an aggregation, we don't validate that it only selects from a single table (no joins). This adds that validation.

### Test Plan

Added a metric test `test_raise_on_joins`.

- [ ] PR has an associated issue: #
- [ ] `make check` passes
- [ ] `make test` shows 100% unit test coverage

### CODE REVIEW

File: datajunction-server/datajunction_server/models/node.py [hunk 1]
```diff
@@ -739,20 +740,34 @@ def check_metric(self):
...
 
+        # must query from a single table
+        if (
+            len(tree.select.from_.relations) != 1
+            or tree.select.from_.relations[0].extensions
+        ):
+            raise DJInvalidInputException(
+                http_status_code=HTTPStatus.BAD_REQUEST,
+                message=f"Metric {self.name} selects from more than one table, "
+                "should only select from a single table",
```
Comment (shangyian) [inline datajunction-server/datajunction_server/models/node.py:768]: Should this message use the `table` terminology or would `node` be clearer? (Not sure as I'm not even sure users like `node`, but it fits well with what we call them)

File: datajunction-server/tests/api/metrics_test.py [hunk 2]
```diff
@@ -1073,7 +1073,69 @@ def test_raise_on_multiple_expressions(client_with_examples: TestClient):
...
+    response = client_with_examples.post(
+        "/nodes/source/",
+        json={
+            "columns": [
+                {
+                    "name": "counts",
+                    "type": "struct<a string, b bigint>",
+                },
+            ],
+            "description": "Collection of day dreams",
```
Comment (shangyian) [inline datajunction-server/tests/api/metrics_test.py:1114]: hahah

**Review by shangyian — APPROVED** [review]

File: datajunction-server/datajunction_server/models/node.py (hunk 1 above)
Comment (agorajek) [reply datajunction-server/datajunction_server/models/node.py:768]: Agree with @shangyian . I think there is a chance we can win this "node vs table" so I would push on that concept where it fits well. The other DJ dag terms we can still keep under covers, like "bound dimensions" ;)

Comment (samredai) [reply datajunction-server/datajunction_server/models/node.py:768]: Good catch, we should consistently use node so I'll update this.

**Pull request merged**